### PR TITLE
SLING-10733 consolidate script documentation

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/jbake/content/documentation/bundles/scripting.md [hunk 1]
```diff
@@ -67,9 +67,143 @@ Mapping `html` to Thymeleaf 3.0:
...
+  properties in the `maven-bundle-plugin`'s instructions (check [these examples](https://github.com/apache/sling-org-apache-sling-scripting-bundle-tracker-it/tree/master/examples/));
+  6. `mvn clean sling:install`.
+
+### Integration Tests
+
+The integration tests for bundled scripts are provided by the [`org.apache.sling.scripting.bundle.tracker.it`](https://github.com/apache/sling-org-apache-sling-scripting-bundle-tracker-it) project.
+
+## Script resolution order
+
+The [same rules as for servlets][6] are being followed but in addition keep in mind that bundled scripts (as well as servlets) are prefered over resource scripts.
```
Comment (kwin) [inline src/main/jbake/content/documentation/bundles/scripting.md:184]: @karlpauls Is this observation correct? I couldn't find according tests for that....

File: src/main/jbake/content/documentation/bundles/scripting.md [hunk 2]
```diff
@@ -67,9 +67,143 @@ Mapping `html` to Thymeleaf 3.0:
...
+Although traditionally scripts are deployed as content stored in the search paths of a Sling instance, this leaves very little
+room for script evolution in a backwards compatible way. Furthermore, versioning scripts is a difficult process if the only
+mechanism to do this is the `sling:resourceType` property, since consumers (content nodes or other resource types) have then to
+explicitly mention the version expected to be executed.
+
+Scripts should not be considered content, since their only purpose is to actually generate the rendering for a certain content
+structure. They are not consumed by users, but rather by the Sling Engine itself and have very little meaning outside this
+context. As such, scripts should be handled like code:
+
+  1. they _provide an HTTP API_;
```
Comment (oliverlietz) [inline src/main/jbake/content/documentation/bundles/scripting.md:133]: _HTTP API_ is quite generic and should be clarified. _HTTP API_ in what sense? Examples?

File: src/main/jbake/content/documentation/bundles/scripting.md [hunk 3]
```diff
@@ -67,9 +67,143 @@ Mapping `html` to Thymeleaf 3.0:
...
+* `{selectorStringPath}` - The selector string converted to a path, along the lines of `selectorString.replace('.', '/')`. If less selectors are specified in the script name than given in the request, the script will only be taken into consideration if the given selectors are the **first** selectors in the request. This means *sel1/sel2.html.jsp* will be a candidate for the request url */content/test.sel1.sel2.sel3.html* but not for */content/test.sel3.sel1.sel2.html*. So the order of selectors is relevant!
+
+
 ## Bundled Scripts
 
-Scripts may also be provided in OSGi bundles (precompiled or embedded) since [Sling Servlet Resolver 2.7.0](https://github.com/apache/sling-org-apache-sling-servlets-resolver#bundled-scripts).
+Scripts may also be provided in OSGi bundles (precompiled or embedded) since [Sling Servlet Resolver 2.7.0](https://github.com/apache/sling-org-apache-sling-servlets-resolver#bundled-scripts) through the org.apache.sling.servlets.resolver.bundle.tracker API in addition to classical Resource based scripts.
```
Comment (oliverlietz) [inline src/main/jbake/content/documentation/bundles/scripting.md:122]: _Scripts may also be provided in OSGi bundle_ – that was already possible with Bundle Resource (no need to store scripts in JCR) and should be clarified.

File: src/main/jbake/content/documentation/bundles/scripting.md (hunk 2 above)
Comment (kwin) [reply src/main/jbake/content/documentation/bundles/scripting.md:133]: This part was just copied from https://github.com/apache/sling-org-apache-sling-servlets-resolver/blob/master/README.md#bundled-scripts. I am open for suggestions on how to improve...

File: src/main/jbake/content/documentation/bundles/scripting.md (hunk 3 above)
Comment (kwin) [reply src/main/jbake/content/documentation/bundles/scripting.md:122]: Good point, I am gonna clarify

File: src/main/jbake/content/documentation/bundles/scripting.md (hunk 2 above)
Comment (raducotescu) [reply src/main/jbake/content/documentation/bundles/scripting.md:133]: @oliverlietz, a script or collection of scripts create an HTTP API once we map a certain content resource to the `resourceType` that would call those scripts. The HTTP API is defined by:
1. the resource (endpoint) that is accessible:
  - the resource path + at least the default extension
  - the resource path + any combination of allowed selectors
2. for scripts which can handle more than just the `GET` request, you're also creating a RESTful API

File: src/main/jbake/content/documentation/bundles/scripting.md (hunk 3 above)
Comment (kwin) [reply src/main/jbake/content/documentation/bundles/scripting.md]: Done in https://github.com/apache/sling-site/pull/67/commits/093fea877b20e019173719a3bc3a013aaeecbc00.

**Pull request merged**